truffleHog: init at 2.0.91

bhipple opened the pull request:
This commit adds the python3 application truffleHog, which is a stand-alone tool
that scans a git repo for unencrypted passwords.

This depends on a newer GitPython, which depends on a new major version of
gitdb, which depends on a new major version of smmap2, so I've packaged those
as well.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

bhipple commented on pkgs/tools/security/truffleHog/default.nix:
```diff
@@ -0,0 +1,38 @@
+    # Relax overly restricted version constraint
+    postPatch = ''
+      sed -i "s/GitPython == 2.1.1/GitPython >= 2.1.1/" setup.py
+    '';
```
CC @FRidh is there a better way to loosen overly conservative semver bounds?  I did build all of these packages, and verified that truffleHog is working fine with the newer GitPython.

dotlambda replied:
Sadly, there isn't. But using
```
sed -i "s/GitPython ==/GitPython >=/" setup.py
```
could reduce the pain upon updating.
Also, consider using `substituteInPlace`.

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40391&attempt_id=a2c46a3e-bb57-4f2f-86ee-1c92d1dba883)

Attempted: truffleHog

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed truffleHog-2.0.91
/build/truffleHog-2.0.91
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/0nmcba3cl2vgd51gqrvwy65rfl51krgv-truffleHog-2.0.91
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/0nmcba3cl2vgd51gqrvwy65rfl51krgv-truffleHog-2.0.91/lib  /nix/store/0nmcba3cl2vgd51gqrvwy65rfl51krgv-truffleHog-2.0.91/bin
patching script interpreter paths in /nix/store/0nmcba3cl2vgd51gqrvwy65rfl51krgv-truffleHog-2.0.91
checking for references to /build in /nix/store/0nmcba3cl2vgd51gqrvwy65rfl51krgv-truffleHog-2.0.91...
wrapping `/nix/store/0nmcba3cl2vgd51gqrvwy65rfl51krgv-truffleHog-2.0.91/bin/trufflehog'...
/nix/store/0nmcba3cl2vgd51gqrvwy65rfl51krgv-truffleHog-2.0.91
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40391&attempt_id=8091dc62-5613-44cb-b908-a53b2e2ffb0d)

Attempted: truffleHog

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed truffleHog-2.0.91
/build/truffleHog-2.0.91
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/idgw64adsh2mj9i7p410pp13z5jpkh28-truffleHog-2.0.91
strip is /nix/store/gp7fylxwn18b7pl2c18ks89hsiaxyfvf-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/idgw64adsh2mj9i7p410pp13z5jpkh28-truffleHog-2.0.91/lib  /nix/store/idgw64adsh2mj9i7p410pp13z5jpkh28-truffleHog-2.0.91/bin
patching script interpreter paths in /nix/store/idgw64adsh2mj9i7p410pp13z5jpkh28-truffleHog-2.0.91
checking for references to /build in /nix/store/idgw64adsh2mj9i7p410pp13z5jpkh28-truffleHog-2.0.91...
wrapping `/nix/store/idgw64adsh2mj9i7p410pp13z5jpkh28-truffleHog-2.0.91/bin/trufflehog'...
/nix/store/idgw64adsh2mj9i7p410pp13z5jpkh28-truffleHog-2.0.91
```
</p></details>

dotlambda commented:
Would you mind moving `GitPython` and `gitdb2` out of `python-packages.nix` and creating seperate commits for those?

dotlambda commented on pkgs/tools/security/truffleHog/default.nix:
```diff
@@ -0,0 +1,38 @@
+    buildInputs = [ ];
+    propagatedBuildInputs = [ GitPython truffleHogRegexes ];
+
+    doCheck = false;
```
Add a comment why tests are disabled.

dotlambda commented on pkgs/tools/security/truffleHog/default.nix:
```diff
@@ -0,0 +1,38 @@
+      sed -i "s/GitPython == 2.1.1/GitPython >= 2.1.1/" setup.py
+    '';
+
+    buildInputs = [ ];
```
drop this line

dotlambda commented on pkgs/top-level/all-packages.nix:
```diff
@@ -21110,6 +21110,8 @@ with pkgs;
 
   tup = callPackage ../development/tools/build-managers/tup { };
 
+  truffleHog = python3.pkgs.callPackage ../tools/security/truffleHog { };
```
Please pass `python3` or `python3Packages` as an argument instead. This will allow overriding of specific dependencies' versions if necessary.

bhipple replied:
Like this?
`  truffleHog = pythonPackages.callPackage ../tools/security/truffleHog { };`

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -2843,20 +2843,39 @@ in {
 
-    buildInputs = with self; [ mock nose ];
-    propagatedBuildInputs = with self; [ gitdb ];
+    buildInputs = with self; [ mock nose ddt ];
```
`checkInputs`

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -14291,6 +14310,24 @@ in {
+      sha256 = "1hvn28p3zvxa98sbi9lrqvv2ps4q284j4jq9a619zw0m7yv0sly7";
+    };
+
+    buildInputs = with self; [ nosexcover ];
```
`checkInputs`

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -2843,20 +2843,39 @@ in {
     };
+  };
+
+  gitdb2 = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -2843,20 +2843,39 @@ in {
+    };
   };
 
   GitPython = buildPythonPackage rec {
```
Can you also move this while you're at it.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -14291,6 +14310,24 @@ in {
     };
   };
 
+  smmap2 = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

bhipple committed:
smmap2: init at 2.0.3

This commit adds smmap2, which is a separate pypi package from the existing
smmap 0.9.0, though they both come from the same repo. It also moves smmap into
the new python modules location.

bhipple committed:
gitdb2: init at 2.0.3

Also move gitdb and gitdb2 to the new python module structure.

bhipple committed:
GitPython: 2.0.8 -> 2.1.9

Also moves GitPython into the newer python module folder structure.

bhipple committed:
truffleHog: init at 2.0.91

This commit adds the python3 application truffleHog, which is a stand-alone tool
that scans a git repo for unencrypted passwords.

This depends on a newer GitPython, which depends on a new major version of
gitdb, which depends on a new major version of smmap, so I've packaged those
as well in the preceding commits.

bhipple commented:
Thx guys; the newer python module structure looks a lot cleaner and nicer!  I've refactored this into four commits: the first one moves smmap2, the second gitdb2, the third GitPython, and the last one truffleHog.

smmap2 and gitdb2 come from the same repo as smmap and gitdb, but they're different packages on pypi and have different import names.  It looks like GitPython is the only one that was using these, at least within nixpkgs; do we generally drop older major versions or leave them around in case someone else wants them?

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40391&attempt_id=b5eed879-0a62-4743-bc28-9f0e889a3b92)

Attempted: truffleHog

<details><summary>Partial log (click to expand)</summary><p>

```
Installing collected packages: truffleHog
Successfully installed truffleHog-2.0.91
/build/truffleHog-2.0.91
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/biz06ll6li8alshxhmy05442qw0xnfp4-truffleHog-2.0.91
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/biz06ll6li8alshxhmy05442qw0xnfp4-truffleHog-2.0.91/lib  /nix/store/biz06ll6li8alshxhmy05442qw0xnfp4-truffleHog-2.0.91/bin
patching script interpreter paths in /nix/store/biz06ll6li8alshxhmy05442qw0xnfp4-truffleHog-2.0.91
checking for references to /build in /nix/store/biz06ll6li8alshxhmy05442qw0xnfp4-truffleHog-2.0.91...
wrapping `/nix/store/biz06ll6li8alshxhmy05442qw0xnfp4-truffleHog-2.0.91/bin/trufflehog'...
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40391&attempt_id=5265a0a8-9adb-468c-8cd2-bd334ae8d10b)

Attempted: truffleHog

<details><summary>Partial log (click to expand)</summary><p>

```
Successfully installed truffleHog-2.0.91
/build/truffleHog-2.0.91
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/3jd06ccjdiwr03b9qi2wg5p2g9kfwf42-truffleHog-2.0.91
strip is /nix/store/gp7fylxwn18b7pl2c18ks89hsiaxyfvf-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/3jd06ccjdiwr03b9qi2wg5p2g9kfwf42-truffleHog-2.0.91/lib  /nix/store/3jd06ccjdiwr03b9qi2wg5p2g9kfwf42-truffleHog-2.0.91/bin
patching script interpreter paths in /nix/store/3jd06ccjdiwr03b9qi2wg5p2g9kfwf42-truffleHog-2.0.91
checking for references to /build in /nix/store/3jd06ccjdiwr03b9qi2wg5p2g9kfwf42-truffleHog-2.0.91...
wrapping `/nix/store/3jd06ccjdiwr03b9qi2wg5p2g9kfwf42-truffleHog-2.0.91/bin/trufflehog'...
/nix/store/3jd06ccjdiwr03b9qi2wg5p2g9kfwf42-truffleHog-2.0.91
```
</p></details>

FRidh commented on pkgs/development/python-modules/smmap/default.nix:
```diff
@@ -0,0 +1,31 @@
+  };
+}
+
+# smmap2 = buildPythonPackage rec {
```
old, needs to be removed

bhipple replied:
Ah thanks, forgot to remove that; I had initially thought about putting them both in one file, since they come from the same git repo and share the same meta info, but decided against it to keep the consistent directory name -> python import name. 

Removed.

FRidh commented on pkgs/top-level/all-packages.nix:
```diff
@@ -21110,6 +21110,8 @@ with pkgs;
 
   tup = callPackage ../development/tools/build-managers/tup { };
 
+  truffleHog = pythonPackages.callPackage ../tools/security/truffleHog { };
```
Can you keep the attribute entirely lowercase, and the same for the folder of the package.

FRidh replied:
No, this should simply be `callPackage`, and the `trufflehog` expressions then has a parameter `python` instead of the individual Python packages and functions that are needed.

bhipple replied:
Lowercased and swapped to `callPackage`; thanks!  Should I reword the commit message to `trufflehog: init at 2.0.91`, or leave it consistent with the pypi name?
